### PR TITLE
Don't run test if exception was thrown in _before of a module

### LINKED ISSUE #4191: Exception in _before of module does not stop execution of the test
Command:
```
php codecept run tests/web/WebDriverTest.php:testClickHashLink
```

Output:
```
Web Tests (1) -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
E WebDriverTest: Click hash link                                                                                                                                                                                           
E WebDriverTest: Click hash link (0.01s)                                                                                                                                                                                   
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                                                                                                                           
                                                                                                                                                                                                                           
Time: 315 ms, Memory: 14.00MB                                                                                                                                                                                              
                                                                                                                                                                                                                           
There were 2 errors:                                                                                                                                                                                                       
                                                                                                                                                                                                                           
---------                                                                                                                                                                                                                  
1) WebDriverTest: Click hash link                                                                                                                                                                                          
 Test  tests/web/WebDriverTest.php:testClickHashLink                                                                                                                                                                       
                                                                                                                                                                                                                           
  [ConnectionException] Can't connect to Webdriver at http://127.0.0.1:4444/wd/hub. Please make sure that Selenium Server or PhantomJS is running.                                                                         
                                                                                                                                                                                                                           
#1  /.../codeception/src/Codeception/Module/WebDriver.php:331                                                                                                                                            
#2  Codeception\Subscriber\Module->before                                                                                                                                                                                  
#3  /.../codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:174
#4  /.../codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:43
#5  /.../codeception/codecept:36

---------
2) WebDriverTest: Click hash link
 Test  tests/web/WebDriverTest.php:testClickHashLink
                                                   
  [Error] Call to a member function get() on null  
                                                   
#1  /.../codeception/tests/web/WebDriverTest.php:768
#2  WebDriverTest->testClickHashLink
#3  /.../codeception/codecept:36

ERRORS!
Tests: 1, Assertions: 0, Errors: 2.
```

Expected output:
```
Web Tests (1) -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------                                                                                                                                                                                           
E WebDriverTest: Click hash link (0.01s)                                                                                                                                                                                   
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                                                                                                                           
                                                                                                                                                                                                                           
Time: 315 ms, Memory: 14.00MB                                                                                                                                                                                              
                                                                                                                                                                                                                           
There were 1 error:                                                                                                                                                                                                       
                                                                                                                                                                                                                           
---------                                                                                                                                                                                                                  
1) WebDriverTest: Click hash link                                                                                                                                                                                          
 Test  tests/web/WebDriverTest.php:testClickHashLink                                                                                                                                                                       
                                                                                                                                                                                                                           
  [ConnectionException] Can't connect to Webdriver at http://127.0.0.1:4444/wd/hub. Please make sure that Selenium Server or PhantomJS is running.                                                                         
                                                                                                                                                                                                                           
#1  /.../codeception/src/Codeception/Module/WebDriver.php:331                                                                                                                                            
#2  Codeception\Subscriber\Module->before                                                                                                                                                                                  
#3  /.../codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:174
#4  /.../codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:43
#5  /.../codeception/codecept:36

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

This is a follow up to #3653

### PR DESCRIPTION
**Pull request by Naktibalda**
I fixed that for all test formats except Unit - it has to be fixed in PHPUnit.
One more reason to use Codeception formats.

Closes #4191

### CODE REVIEW

File: src/Codeception/Test/Test.php [hunk 1]
```diff
@@ -70,20 +70,18 @@
         $time = 0;
         $e = null;
         
-        try {
-            $result->startTest($this);
-        } catch (\Exception $er) {
-            // failure is created: not a user's test code error so we don't need detailed stacktrace
-            $this->testResult->addError($this, new \PHPUnit_Framework_AssertionFailedError($er->getMessage()), 0);
-            $this->ignored = true;
```
Comment (DavertMik) [inline src/Codeception/Test/Test.php]: not sure why the test was executed even when it was marked as ignored :confused:

**Comment by DavertMik** [comment]
Thanks. Very nice fix

**Comment by nebosklondev** [comment]
I run 26 tests. And if the second does not pass, then all other tests are not performed. It's horrible.

**Comment by Naktibalda** [comment]
I know, it is fixed already and will be released soon.
Please downgrade to 2.2.10 if 2.2.11 is unusable.

**Pull request merged**